### PR TITLE
Added some more potentially robust ways to do learning rate tuning

### PR DESCRIPTION
**Pull request by varchasgopalaswamy**
## What does this PR do?

As discussed in https://github.com/Lightning-AI/pytorch-lightning/issues/1767, the learning rate tuner suggests the minimum gradient of the loss via `torch.argmin(torch.gradient(losses)[0])`. This makes it a little sensitive to a noisy loss vs LR landscape. I added a few methods discussed [here](https://forums.fast.ai/t/automated-learning-rate-suggester/44199/30), and found they worked pretty well for my use cases (see example below, where gradient-based estimation picks out a very low LR). 

image

Defaults are set so no breaking changes are introduced.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Comment by Borda** [comment]
cc: @SkafteNicki
